### ISSUE TITLE
Ammo tracking doesn't work in newer Foundry versions

### ISSUE THREAD

**Issue by MajsterEmski** [fvtt-fria-ligan/morkborg-foundry-vtt#185]
In Foundry v.12, with ammo tracking enabled and an appropriate ammo item linked in the Violence tab, an error occurs: upon attacking, dice are rolled for hit _(and damage in the case of a succesful hit, so underlying logic works fine)_, however, no **Ranged Attack** message is sent to the chat and linked ammo count is not reduced. The console logs the following error:
`Error: You are accessing the MBItem "data" field of which was deprecated in v10 and replaced with "system". Continued usage of pre-v10 ".data" paths is no longer supported"`
Everything works as intended when no ammo item is linked in the Violence tab — it's probably only the `decrementWeaponAmmo` function in **attack.js** that's affected.

**Comment by ElderRune75** [comment]
No result messages are showing in the chat when attacking with ranged weapons when "track ammo is turned on in Settings".
This goes for Bow, short bow and crossbow with bolts and arrow items as "ammo".
What can i do to make this work again?
Foundry VTT v.12 b.330
Mörk Borg Module v 4.0.1
The messages work with the "track ammo" turned off with "Hit automation" on and off.
As soon as the ranged weapon has a specific ammo-type selected or "track ammo" is selected in the settings - the messages stop working.
![ammo](https://github.com/user-attachments/assets/e833bf2c-f7d6-4a02-9f88-b4de4c757509)

1/3 is attacking with no ammo selected and "Hit automation" turned off.
2/3 is selecting bolts-ammo for equipped crossbow and "Hit automation" turned off.
3/3 is attacking with bolts-ammo selected for equipped crossbow and "Hit automation" turned off.

**Comment by ElderRune75** [comment]
MajsterEmski did you get this fixed?

**Comment by MajsterEmski** [comment]
No, sorry—I'm not sufficiently savvy when it comes to Foundry & JavaScript, yet. I might get to it once some life stuff gets sorted out. For the time being, I merely reported the bug.
